src/main/java/bio/terra/cli/apps/DockerAppsRunner.java

```diff
@@ -107,6 +107,12 @@ public void runToolCommand(
     // check that the current workspace is defined
     workspaceContext.requireCurrentWorkspace();
 
+    // substitute any Terra references in the command
+    // also add the Terra references as environment variables in the container
+    Map<String, String> terraReferences = buildMapOfTerraReferences();
+    command = replaceTerraReferences(terraReferences, command);
```

Comment:
I guess I'm wondering why env variables aren't enough on their own here? Does the main Docker command not allow for env variable substitution for some reason? If there's some reference source on why that's the case, a short comment calling out to the docs might help here.

Reply:
I think the env var substitution is a bash/shell thing, so it's not that Docker doesn't support it, it's that we need to bypass the bash env var substitution on the host machine so that the Terra CLI can handle it.

For example:
`
host-machine> env
HOST_ENV_VAR=123
host-machine> terra nextflow run $HOST_ENV_VAR
`
causes the Terra CLI Java code to see the command arguments `nextflow run 123` only. It doesn't know about any env var substitution that's already happened.

I wanted to bypass the env var substitution on the host machine so I can do it in the Terra CLI. So I used a slightly different syntax:
`
host-machine> terra nextflow run {HOST_ENV_VAR}
`
causes the Terra CLI Java code to see the command arguments `nextflow run {HOST_ENV_VAR}` and then I can do the substitution in the Terra CLI before passing the modified command string to Docker.